### PR TITLE
feat: resolve security concerns with Clarity integration

### PR DESCRIPTION
**Pull request by thanhdanh**
## Summary
Follow these [security concerns](https://github.com/ExodusMovement/exodus-mobile/pull/19513#issuecomment-2119927319) and fix:

- Add docs about jsonParam.
- Exclude the test folder by reverting `.gitignore`
-  Removed the unused `requestBuffer`  in MoneroAsyncTask.java:87
- Removed the validation of the url parameter in Objective-C.
- Close the connection in Java code.
- Adding checks that `malloc` actually returned not null in C++. The size of the response is larger, so this check helps prevent undefined behavior and potential security vulnerabilities such as buffer overflows or null pointer dereferences.

### CODE REVIEW

File: ios/RNFastCrypto.m [hunk 1]
```diff
@@ -111,11 +111,6 @@ + (void) handleDownloadFromClarityAndProcess:(NSString*) method
 
     NSString *addr = jsonParams[@"url"];
     NSURL *url = [NSURL URLWithString:addr];
-    if (!url) {
```
Comment (mbaraniak-exodus) [inline ios/RNFastCrypto.m:114]: Do we have URL validation elsewhere ?

**Review by mbaraniak-exodus — COMMENTED** [review]
uACK

**Comment by feri42** [comment]
Please update the PR title to reflect the code changes.

File: ios/RNFastCrypto.m (hunk 1 above)
Comment (thanhdanh) [reply ios/RNFastCrypto.m:114]: Nope, we didn't check it, but it is sent from js call, so it's never empty.

**Review by feri42 — APPROVED** [review]
utACK

**Pull request merged**